adds new content section (#24)
ecoacoustics/website: content/resources/help-centre/software-tips/path.md

---
title: The PATH variable
---

## What is the "PATH"?

If you have been following the lessons and help guides available on this
website, and installing software such as FFmpeg and AP (AnalysisPrograms) then
you have probably seen the term `PATH` by now. 

> The `PATH` variable is a system wide variable (known as en environment variable)

Reviewer: Not strictly true. 

Windows has two path environment variables that get merged: a system one and a per-user one.

```suggestion
The `PATH` variable an environment variable. Environment variables are values that 
```

that any program can access. The `PATH` variable contains a list of folders that
your computer can check when it searches for programs to run.

> In simple terms, if we want to use a command from a particular software in our

Reviewer: ```suggestion
In simple terms, if we want to use a program in our
```

shell (see the [command line]({{% ref "command-line.md" %}}) guide), we have two

> choices. We can specify the full path to the `.exe` file, and then write our

Reviewer: ```suggestion
choices. We can specify the full path to the program (often referred to as a binary,  usually has an extension like `.exe` on Windows) file, and then write our
```

code:

```ps
> C:\Users\Anthony\AP\AnalysisPrograms.exe 

> ```

Reviewer: since AP has a linux installer you can show the linux variant here too

But this would be annoying to write out every time. Instead, if our `PATH`
variable includes the AP folder, we can run AP commands without specifying this
full path! 

```ps
> AnalysisPrograms.exe
```

## How to check the PATH variable

We can quickly check if a program has been successfully added to our `PATH` or
not. Open up a shell (such as PowerShell on Windows, or terminal on
Linux/macOS), and type a command for the program we are interested in testing. 

```ps
> FFmpeg
```

If the command successfully returns some kind of output about the package, then
it is available on your `PATH`.  

> 

Reviewer: you probably want to show them how to see path, 

`printenv | grep PATH` in bash and ` echo $env:Path` in pwsh

## Adding to PATH

If you use a [package manager]({{% ref "package-manager.md" %}}) such as
[chocolatey](https://chocolatey.org/) on Windows, [homebrew](https://brew.sh/)
on macOS or [snap](https://snapcraft.io/) on Linux, then packages you install
with those commands should be automatically added to your `PATH` variable. This
saves you having to manually change the `PATH` variable. The automatic installer

> for [AP](https://ap.qut.ecoacoustics.info/basics/installing.html?tabs=windows)

Reviewer: ```suggestion
for [AP](https://ap.qut.ecoacoustics.info/basics/installing.html)
```

6. Then click OK or close all windows.
7. You will have to restart any programs for which you want to see the new value
{{% /tab %}}
{{% tab "macOS" %}}
1. Find where the program is installed on your computer. This will be a
   directory (folder) where the .exe resides.
2. Open or create your ~/.profile file
3. Add the following line to the end:

`PATH=$PATH:<REPLACE-ME>`

where you replace the `<REPLACE-ME>` with the directory from step 1.

4. Close and save the file
5. Run the same command in your current shell (PATH=$PATH:<REPLACE-ME>) to see the change take effect immediately (or restart your shell).
{{% /tab %}}
{{% tab "Linux" %}}
1. Find where the program is installed on your computer. This will be a
   directory (folder) where the .exe resides.
2. Open or create your ~/.profile file
3. Add the following line to the end:

`PATH=$PATH:<REPLACE-ME>`

where you replace the `<REPLACE-ME>` with the directory from step 1.

4. Close and save the file
5. Run the same command in your current shell (PATH=$PATH:<REPLACE-ME>) to see the change take effect immediately (or restart your shell).
{{% /tab %}}

> {{% /tabs %}}

Reviewer: so fun fact: there are system wide environment files you can configure for linux (which would be the same as your windows example)
![image](https://user-images.githubusercontent.com/1452968/199008210-778f75b1-8c5c-4bd7-8249-4c6ce8a720e8.png)

and in powershell you can do `$env:Path=$env:Path;<REPLACE-ME>` in your powershell profile as well!
